### ISSUE TITLE
Logging in native app is not working anymore

### ISSUE THREAD

**Issue by asantare** [micronaut-projects/micronaut-core#9850]
### Expected Behavior

logging should work

### Actual Behaviour

nothing is logged except for the banner

### Steps To Reproduce

1. building and running the demo:
```
export GRAALVM_HOME=... # location of the graalvm jdk
export JAVA_HOME=$GRAALVM_HOME
export PATH="$GRAALVM_HOME/bin:$PATH"
./gradlew nativeCompile -Pmicronaut.runtime=netty -PmainClass=com.example.ApplicationKt
build/native/nativeCompile/demo
```
2. invoking the url:
```
curl http://localhost:8080
```

The log "INDEX INVOKED" is expected.
It works if run in non-native:
```
./gradlew run -Pmicronaut.runtime=netty -PmainClass=com.example.ApplicationKt
```

It is working in 4.0.x

### Environment Information

- MacOS M1
- graalvm-community-openjdk-17.0.8+7.1

### Example Application

https://github.com/asantare/micronaut-native-no-logging

### Version

4.1.3

**Comment by asantare** [comment]
using the workaround from #9835 seems to be working but is not a complete fix, the original project where I encountered the error only logs the 'server started' line with this workaround and nothing else afterwards.

**Comment by asantare** [comment]
I did a second check on the original project and yes, the workaround in #9835 fixes it

**Comment by sdelamo** [comment]
we are reverting logback to 1.4.8 in Micronaut Framework 4.1.1

**Comment by sdelamo** [comment]
This should be fixed via #9857
